src/backend/replication/basebackup.c

```diff
@@ -1754,6 +1755,8 @@ sendFile(const char *readfilename, const char *tarfilename, struct stat *statbuf
 			for (i = 0; i < cnt / BLCKSZ; i++)
 			{
 				page = buf + BLCKSZ * i;
+				if (basebackup_file_before_verify_hook)
+					page = basebackup_file_before_verify_hook(page, readfilename, blkno + i, BLCKSZ);
```

Comment:
Since the `BLCKSZ` is always passed as the `block_size`, shall we just remove it from the hook signature? I don't think we need to consider the same hook for AO table right now. When it needs to support AO table, the signature will be changed and break the ABI anyway. We should keep it simple first.

Except this, the PR looks good to me.